spring-cloud-context/src/test/java/org/springframework/cloud/context/named/NamedContextFactoryTests.java

```diff
@@ -62,8 +61,38 @@ public void testChildContexts() {
 		assertThat("bar context wasn't closed", barContext.isActive(), is(false));
 	}
 
+	@Test
+	public void testEagerlyCreateContexts() {
+		long time1 = lazyGetFromContext();
+		long time2 = cacheAndGetBeanFromContext();
+		assertThat(time2, lessThan(time1));
```

Comment:
I worry that this is not a deterministic test. A more valid test would check when creation was done. Not sure how you'd set that up though.